hw/mcu/dialog/da1469x/src/da1469x_lpclk.c

```diff
@@ -57,11 +64,18 @@ da1469x_lpclk_register_cmac_cb(da1469x_lpclk_cb *cb)
 void
 da1469x_lpclk_init(void)
 {
+#if MYNEWT_VAL_CHOICE(MCU_LPCLK_SOURCE, XTAL32K)
     static struct hal_timer lpclk_settle_tmr;
-
     da1469x_clock_lp_xtal32k_enable();
-
     os_cputime_timer_init(&lpclk_settle_tmr, da1469x_lpclk_settle_tmr_cb, NULL);
     os_cputime_timer_relative(&lpclk_settle_tmr,
                               MYNEWT_VAL(MCU_CLOCK_XTAL32K_SETTLE_TIME_MS) * 1000);
+#endif
+#if MYNEWT_VAL_CHOICE(MCU_LPCLK_SOURCE, RCX)
+    static struct hal_timer lpclk_settle_tmr;
+    da1469x_clock_lp_rcx_enable();
+    os_cputime_timer_init(&lpclk_settle_tmr, da1469x_lpclk_settle_tmr_cb, NULL);
+    os_cputime_timer_relative(&lpclk_settle_tmr,
+                              MYNEWT_VAL(MCU_CLOCK_RCX_SETTLE_TIME_MS) * 1000);
```

Comment:
I'd just switch to RCX immediately after XTAL32M is settled - we need then to add RCX calibration code, but more changes are required since RCX has different frequency than XTAL32K and our lpm code is not ready for this so RCX will not be reliable anyway, need to see what changes are required